cvat/apps/engine/cron.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+import os
+import os.path as osp
+from datetime import timedelta
+from functools import wraps
+from threading import Event, Thread
+from time import sleep
+from typing import Callable
+
+from django.conf import settings
+from django.utils import timezone
+from rq import get_current_job
+
+from cvat.apps.dataset_manager.util import ExportCacheManager, get_export_cache_lock
+from cvat.apps.dataset_manager.views import (
+    EXPORT_CACHE_LOCK_ACQUISITION_TIMEOUT,
+    EXPORT_CACHE_LOCK_TTL,
+    get_export_cache_ttl,
+    log_exception,
+)
+from cvat.apps.engine.log import ServerLogManager
+
+logger = ServerLogManager(__name__).glob
+
+
+def suppress_exceptions(func: Callable[[CleanupExportCacheThread], None]):
+    @wraps(func)
+    def wrapper(self: CleanupExportCacheThread):
+        try:
+            func(self)
+        except Exception as ex:
+            self.set_exception(ex)
+
+    return wrapper
+
+
+def clear_export_cache(file_path: str) -> None:
+    with get_export_cache_lock(
+        file_path,
+        block=True,
+        acquire_timeout=EXPORT_CACHE_LOCK_ACQUISITION_TIMEOUT,
+        ttl=EXPORT_CACHE_LOCK_TTL,
+    ):
+        parsed_filename = ExportCacheManager.parse_file_path(file_path)
+        cache_ttl = get_export_cache_ttl(parsed_filename.instance_type)
+
+        if timezone.now().timestamp() <= osp.getmtime(file_path) + cache_ttl.total_seconds():
+            logger.debug(f"Export cache file {file_path!r} was recently accessed")
+            return
+
+        os.remove(file_path)
+        logger.debug(f"Export cache file {file_path!r} was successfully removed")
```

Comment:
You should either move this into the `dataset_manager` app or move test from that app into `engine`. Currently we have some mix with no clear responsibility separation between components.

Reply:
Moved `cron.py` into the `dataset_manager` app. I don't really think that it is the purpose of this PR to refactor the functionality layout.